ui/src/api-logic/mappers/repos.ts

```diff
@@ -0,0 +1,71 @@
+import { RepoDataPropsT, RepoDataStatusT } from 'src/@types'
+import { getTimeAgoFromNow } from 'src/utils'
+import { GetReposQuery } from '../graphql/generated/schema'
+
+const mapToRepoData = (data: GetReposQuery | undefined): Array<RepoDataPropsT> => {
+  const mappedData: Array<RepoDataPropsT> = []
+
+  data?.repos?.nodes.forEach((r) => {
+    let repoInfo: RepoDataPropsT = {
+      name: r?.repo.replace('https://github.com/', '') || '',
+      lastUpdate: getTimeAgoFromNow(new Date(r?.createdAt)) || '',
+      lastSync: '',
+      type: r?.isGithub ? 'github' : 'other',
+      tags: [],
+      status: [],
+    }
+
+    repoInfo.status = getStatusFromRepo(r, repoInfo)
+    mappedData.push(repoInfo)
+  })
+
+  return mappedData
+}
+
+const getStatusFromRepo = (r: any, repoInfo: RepoDataPropsT): Array<RepoDataStatusT> => {
+  const syncTypes = r?.repoSyncs.nodes.map((st: any) => {
+    const syncObj: Record<string, any> = {
+      type: st?.syncType,
+    }
+
+    st?.repoSyncQueues.nodes.forEach((ls: any) => {
+      syncObj.status = ls?.status || ''
+      syncObj.lastSync = ls?.createdAt || ''
+    })
+
+    return syncObj
+  })
+
+  const mapSyncs = new Map()
+  syncTypes?.forEach((st: any) => {
+    const status = mapToRepSyncStateT(st.status)
+    let statusCount = mapSyncs.get(status)
+    mapSyncs.set(status, statusCount ? ++statusCount : 1)
+  })
+
+  const mappedSyncs: Array<RepoDataStatusT> = []
+  mapSyncs?.forEach((value, key) => {
+    mappedSyncs.push({ type: key, count: value })
+  })
+  repoInfo.lastSync = getTimeAgoFromNow(new Date(syncTypes[0].lastSync))
+
+  return mappedSyncs
+}
+
+const mapToRepSyncStateT = (status: string) => {
+  let newStatus
+  switch (status) {
+    case 'DONE':
+      newStatus = 'succeeded'
+      break
+    case 'RUNNING':
+      newStatus = 'running'
+      break
+    default:
+      newStatus = 'failed'
+      break
+  }
+  return newStatus
+}
```

Comment:
```suggestion
const mapToRepoSyncStateT = (status: string) => {
  switch (status) {
    case 'DONE':
      return 'succeeded'
    case 'RUNNING':
      return 'running'
    default:
      return 'failed'
  }
}
```

Just a suggestion to be a bit more succinct!

Reply:
also see the change in the function name (`mapToRepSyncStateT` -> `mapToRepoSyncStateT` - I assume the missing `o` is a typo?)

Reply:
Comments addressed here [5c8df16](https://github.com/mergestat/fuse/pull/74/commits/5c8df164d40cc06bd3739a0f8f4d93647728b502)